LnkParse3/decorators.py

```diff
@@ -0,0 +1,111 @@
+from datetime import datetime
+from datetime import timezone
+from struct import unpack
+import functools
+from LnkParse3.lnk_exception import lnk_exception
+
+
+def must_be(expected):
+    def outer(func):
+        @functools.wraps(func)
+        def inner(self, *args, **kwargs):
+            result = func(self, *args, **kwargs)
+
+            # FIXME: delete
+            return result
+
+            if result != expected:
+                error = "%s must be %s: %s" % (func.__name__, expected, result)
+                raise lnk_exception(error)
+
+            return result
+
+        return inner
+
+    return outer
+
+
+def uuid(func):
+    @functools.wraps(func)
+    def inner(self, *args, **kwargs):
+        binary = func(self, *args, **kwargs)
+
+        # FIXME: delete
+        return binary.hex()
+
+        # UUID variants
+        # https://docs.microsoft.com/en-us/openspecs/windows_protocols/ms-dtyp/49e490b8-f972-45d6-a3a4-99f924998d97
+        d1, d2, d3 = unpack("<LHH", binary[0:8])
+        d4, d51, d52 = unpack(">HHI", binary[8:16])
+        d5 = (d51 << 16) | d52
+
+        uuid = "%08X-%04X-%04X-%04X-%012X" % (d1, d2, d3, d4, d5)
+
+        return uuid
+
+    return inner
+
+
+def filetime(func):
+    @functools.wraps(func)
+    def inner(self, *args, **kwargs):
+        binary = func(self, *args, **kwargs)
+        nanosec = unpack("<q", binary)[0]
+
+        # FIXME: zero width string?
+        if nanosec == 0:
```

Comment:
Is this a problem?

Reply:
While reading JSON file, I would expect corrupted data as `null`.  I'm also not sure whether zero value should be considered as corrupted.  Do you have any references of boundary value for filetime/dostime?
There is another representation `"Invalid time"`.  I think we should normalize what means corrupted and what is not.

Reply:
I don't have any references. I think I would consider zero as a valid date, but not sure.

Reply:
OK, let's keep it as is.  zero value == invalid.